bottlecap/src/logs/processor.rs

```diff
@@ -0,0 +1,339 @@
+                .. // TODO: check if we could do something with this metrics: `initialization_type` and `phase`
+            } => {
+                let rv = runtime_version.unwrap_or("?".to_string()); // TODO: check what does containers display
+                let rv_arn = runtime_version_arn.unwrap_or("?".to_string()); // TODO: check what do containers display
+                log.message.message = format!("INIT_START Runtime Version: {} Runtime Version ARN: {}", rv, rv_arn);
+                log.message.status = "info".to_string();
+
+                Ok(log)
+            },
+            TelemetryRecord::PlatformStart {
+                request_id,
+                version,
+            } => {
+                let version = version.unwrap_or("$LATEST".to_string());
+                log.message.message =
+                    format!("START RequestId: {} Version: {}", request_id, version);
+                log.message.lambda.request_id = request_id;
+                log.message.status = "info".to_string();
+
+                Ok(log)
+            },
+            TelemetryRecord::PlatformRuntimeDone { request_id , .. } => {  // TODO: check what to do with rest of the fields
+                log.message.message = format!("END RequestId: {}", request_id);
+                log.message.lambda.request_id = request_id;
+                log.message.status = "info".to_string();
+                Ok(log)
+            },
+            TelemetryRecord::PlatformReport { request_id, metrics, .. } => { // TODO: check what to do with rest of the fields
+                let mut message = format!(
+                    "REPORT RequestId: {} Duration: {} ms Billed Duration: {} ms Memory Size: {} MB Max Memory Used: {} MB",
+                    request_id,
+                    metrics.duration_ms,
+                    metrics.billed_duration_ms,
+                    metrics.memory_size_mb,
+                    metrics.max_memory_used_mb,
+                );
+
+                let init_duration_ms = metrics.init_duration_ms;
+                if let Some(init_duration_ms) = init_duration_ms {
+                    message = format!("{} Init Duration: {} ms", message, init_duration_ms)
+                }
+
+                log.message.message = message;
+                log.message.lambda.request_id = request_id;
+                log.message.status = "info".to_string();
+
+                Ok(log)
+            },
+            // TODO: PlatformInitRuntimeDone
+            // TODO: PlatformInitReport
+            // TODO: PlatformExtension
+            // TODO: PlatformTelemetrySubscription
+            // TODO: PlatformLogsDropped
+            _ => Err("Unsupported event type".into()),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::telemetry::events::{
+        InitPhase, InitType, ReportMetrics, RuntimeDoneMetrics, Status,
+    };
+
+    use super::*;
+
+    use chrono::{TimeZone, Utc};
+
+    macro_rules! process_tests {
```

Comment:
I'm impressed you've grokked the macro rules stuff so quickly. I'd recommend that this could probably just be a helper function. Maintaining macro rules blocks tends to be a bit of a pain, and the added verbosity isn't that much. Alternatively, you can use something like https://docs.rs/duplicate/latest/duplicate/ to do the same thing. One other thing to consider, if you want a lot of tests here, and the IntakeOutput might change frequently as we add features, you could use a snapshot testing tool like [insta](https://docs.rs/insta/latest/insta/), which can cut down a lot of verbosity in the code.

Reply:
I think I'll address this in another PR to use duplicate/insta on the whole repo, WDYT?